packages/@react-stately/layout/src/TableLayout.ts

```diff
@@ -183,12 +127,13 @@ export class TableLayout<T> extends ListLayout<T> {
     }
   }
 
-  getColumnWidth(node: GridNode<T>) {
+  // used to get the column widths when rendering to the DOM
+  getColumnWidth_(node: GridNode<T>) {
     let colspan = node.colspan ?? 1;
     let width = 0;
     for (let i = 0; i < colspan; i++) {
       let column = this.collection.columns[node.index + i];
-      width += this.columnWidths.get(column.key);
+      width += this.getColumnWidth(column.key);
```

Comment:
I know this code was like this already, but maybe we could change the loop to make it more obvious what it's doing?
```
    for (let i = node.index; i < node.index + colspan; i++) {
      let column = this.collection.columns[i];
      width += this.getColumnWidth(column.key);
```